### PR TITLE
Use env values

### PR DESCRIPTION
**Pull request by arbulu89**
Load the new `env` field from the contract and expand it to the execution dynamic server.

### CODE REVIEW

File: test/support/factory.ex [hunk 1]
```diff
@@ -19,6 +19,14 @@ defmodule Wanda.Factory do
 
   alias Wanda.Results.ExecutionResult
 
+  def map_factory(attrs) do
```
Comment (arbulu89) [inline test/support/factory.ex:22]: I could very well name this `env_factory` to be more specific, as it is only used to build the `env` data

File: lib/wanda/policy.ex [hunk 2]
```diff
@@ -26,14 +26,16 @@ defmodule Wanda.Policy do
...
+         env: env
        }) do
     execution_impl().start_execution(
       execution_id,
       agent_id,
       Enum.map(targets, fn %{agent_id: agent_id, checks: checks} ->
         %Target{agent_id: agent_id, checks: checks}
-      end)
+      end),
+      Map.new(env, fn {key, %{kind: {_, value}}} -> {key, value} end)
```
Comment (nelsonkopliku) [inline lib/wanda/policy.ex:38]: Where does this `kind` come from? :thinking: 
Is it something related to the `random_env_kind` factory?

File: test/support/factory.ex [hunk 3]
```diff
@@ -51,6 +51,33 @@ defmodule Wanda.Factory do
     }
   end
 
+  def env_factory(attrs) do
```
Comment (nelsonkopliku) [inline test/support/factory.ex:54]: If I am understanding it correctly `env_event_factory` is used in tests _closer_ to the integration layer where `ExecutionRequested` event is involved.
While `env_factory` is used in more inner layers.

What other differences are there between `env_factory` `env_event_factory`?
Would a single factory be enough?

**Review by nelsonkopliku — COMMENTED** [review]
Overall looks good and goes in a reasonable direction.
Commented on a couple of doubts.

File: test/support/factory.ex (hunk 3 above)
Comment (arbulu89) [reply test/support/factory.ex:54]: `env_factory` creates data based in our domain (where the env is simply a map).
`env_event_factory` creates the env dataset coming in the `ExecutionRequested` env field.
They are different, as the 2nd is strictly attached to protobuf and it is only used to send and receive messages, not internally.

Maybe I could rename to `execution_requested_env_factory`

**Review by nelsonkopliku — APPROVED** [review]
LGTM

File: lib/wanda/execution/behaviour.ex [hunk 4]
```diff
@@ -9,14 +9,14 @@ defmodule Wanda.Execution.Behaviour do
               execution_id :: String.t(),
               group_id :: String.t(),
               targets :: [Target.t()],
-              env :: %{String.t() => String.t()}
+              env :: map()
```
Comment (fabriziosestito) [inline lib/wanda/execution/behaviour.ex:12]: We could be more precise in the spec, `%{String.t() => boolean() | number() | String.t()}`

**Review by fabriziosestito — CHANGES_REQUESTED** [review]

File: test/policy_test.exs [hunk 5]
```diff
@@ -19,10 +19,7 @@ defmodule Wanda.PolicyTest do
     agent_id = UUID.uuid4()
     env = build(:execution_requested_env)
 
-    expected_env =
-      Enum.reduce(env, %{}, fn {key, %{kind: {_, value}}}, acc ->
-        Map.put(acc, key, value)
-      end)
+    expected_env = Map.new(env, fn {key, %{kind: {_, value}}} -> {key, value} end)
```
Comment (fabriziosestito) [inline test/policy_test.exs:22]: we are repeating the implementation in the test, which makes the test of little value.

**Comment by arbulu89** [comment]
@fabriziosestito Thanks! All the feedback implemented

File: test/policy_test.exs (hunk 5 above)
Comment (nelsonkopliku) [reply test/policy_test.exs]: Ah yeah! Glad you spotted! I overlooked that. My bad :sweat_smile:

File: test/support/factory.ex [hunk 6]
```diff
@@ -51,6 +51,22 @@ defmodule Wanda.Factory do
...
+    count = Map.get(attrs, :count, 5)
+
+    Enum.reduce(0..count, %{}, fn _, acc ->
+      Map.put(acc, Faker.Pokemon.name(), random_env_value())
+    end)
+  end
+
+  defp random_env_value do
```
Comment (fabriziosestito) [inline test/support/factory.ex:62]: defp should always be at the end, after the def functions

**Review by fabriziosestito — APPROVED** [review]
LGTM, just a minor comment

**Pull request merged**